packages/graph/hash_graph/lib/graph/src/knowledge/entity/mod.rs

```diff
@@ -88,16 +95,70 @@ impl PersistedEntityIdentifier {
     }
 }
 
+/// The associated information for 'Link' entities
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, ToSchema)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct LinkEntityMetadata {
+    left_entity_id: EntityId,
+    right_entity_id: EntityId,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    left_order: Option<LinkOrder>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    right_order: Option<LinkOrder>,
+}
+
+impl LinkEntityMetadata {
+    #[must_use]
+    pub const fn new(
+        left_entity_id: EntityId,
+        right_entity_id: EntityId,
+        left_order: Option<LinkOrder>,
+        right_order: Option<LinkOrder>,
+    ) -> Self {
+        Self {
+            left_entity_id,
+            right_entity_id,
+            left_order,
+            right_order,
+        }
+    }
+
+    #[must_use]
+    pub const fn left_entity_id(&self) -> EntityId {
+        self.left_entity_id
+    }
+
+    #[must_use]
+    pub const fn right_entity_id(&self) -> EntityId {
+        self.right_entity_id
+    }
+
+    #[must_use]
+    pub const fn left_order(&self) -> Option<LinkOrder> {
+        self.left_order
+    }
+
+    #[must_use]
+    pub const fn right_order(&self) -> Option<LinkOrder> {
+        self.right_order
+    }
+}
+
 /// The metadata of an [`Entity`] record.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, ToSchema)]
-#[serde(rename_all = "camelCase")]
+// TODO: deny_unknown_fields on other structs
```

Comment:
Similar to my previous comment. I think if we make sure to check for TODOs before the final `dev/links` merge then we can move faster _for now_ by not needing to capture everything in tagged asana tasks, etc.

Reply:
And I suppose we can always look at the diff for newly introduced todos rather than combing through every todo in the codebase

Reply:
Yeah that's what I mean :)